### ISSUE TITLE
security vulnerability found in package axios

### ISSUE THREAD

**Issue by ms-oh** [artilleryio/artillery#2519]
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating it.
\-->

Version info:

```
2.0.6
```

Running this command:

```
npm audit
```

I expected to see this happen:

`Found 0 vulnerabilities`

Instead, this happened:

`Found 1 vulnerability: axios`

Files being used:

```
<none>
```

**Comment by hassy** [comment]
yes, we try to address any reported dependency vulnerabilities in  each
release so we’ll try to upgrade this one in the upcoming release

**Comment by pedroresende** [comment]
any news on this ?

**Comment by dirkluijk** [comment]
Should be fixed by https://github.com/artilleryio/artillery/pull/3387.

**Comment by bonjhhh** [comment]
Hello team, May I kindly request to check this is related? Is this a new issue? Thanks in advance for checking and/or tips on how to fix. 

PS C:\myDirectory\source\artillery-performance-testing> node -v  
v20.17.0
PS C:\myDirectory\source\artillery-performance-testing> npm -v   
10.9.0
PS C:\myDirectory\source\artillery-performance-testing> npm ls axios
artillery-performance-testing@ C:\myDirectory\source\artillery-performance-testing
├─┬ artillery@2.0.21
│ └─┬ posthog-node@2.6.0
│   └── axios@0.27.2
└── axios@1.7.9

PS C:\myDirectory\source\artillery-performance-testing> npm ls posthog-node
artillery-performance-testing@ C:\myDirectory\source\artillery-performance-testing
└─┬ artillery@2.0.21
  └── posthog-node@2.6.0

PS C:\myDirectory\source\artillery-performance-testing> npm audit          
# npm audit report

axios  0.8.1 - 0.27.2
Severity: moderate
Axios Cross-Site Request Forgery Vulnerability - https://github.com/advisories/GHSA-wf5p-g6vw-rhxx
fix available via `npm audit fix --force`
Will install artillery@1.7.4, which is a breaking change
node_modules/artillery/node_modules/axios
  posthog-node  <=1.3.0 || 2.1.0 - 3.1.2
  Depends on vulnerable versions of axios
  node_modules/artillery/node_modules/posthog-node
    artillery  >=1.7.5
    Depends on vulnerable versions of posthog-node
    node_modules/artillery

3 moderate severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

**Comment by bonjhhh** [comment]
Fixed and thank you team! :) 

**PS C:\source\artillery-performance-testing> npm audit**
# npm audit report

axios  0.8.1 - 0.27.2
Severity: moderate
Axios Cross-Site Request Forgery Vulnerability - https://github.com/advisories/GHSA-wf5p-g6vw-rhxx
fix available via `npm audit fix`
node_modules/artillery/node_modules/axios
  posthog-node  <=1.3.0 || 2.1.0 - 3.1.2
  Depends on vulnerable versions of axios
  node_modules/artillery/node_modules/posthog-node
    artillery  1.7.5 - 2.0.21
    Depends on vulnerable versions of posthog-node
    node_modules/artillery

3 moderate severity vulnerabilities

To address all issues, run:
  npm audit fix
**PS C:\source\artillery-performance-testing> npm audit fix**
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'artillery@2.0.22',
npm warn EBADENGINE   required: { node: '>= 22.13.0' },
npm warn EBADENGINE   current: { node: 'v20.17.0', npm: '10.9.0' }
npm warn EBADENGINE }

added 85 packages, removed 3 packages, changed 40 packages, and audited 1047 packages in 2m

**137 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities**
PS C:\source\artillery-performance-testing>
